app/models/concerns/assign_to_current_semester.rb

```diff
@@ -6,6 +6,7 @@ module AssignToCurrentSemester
   end
 
   def assign_to_current_semester
-    self.semester = Semester.current_semester unless semester.present?
+    # TODO: Settings are not initilized yet here
+    # self.semester = Semester.find(Settings.new.current_semester_id)
```

Comment:
yeah. i have a better working version but it still fails because i initialize Settings with "current_semester_id = nil" because there might not be any semesters. which fails the validation